handler/src/main/java/io/netty/handler/logging/LoggingHandler.java

```diff
@@ -317,8 +362,11 @@ private static String formatByteBuf(ChannelHandlerContext ctx, String eventName,
             int rows = length / 16 + (length % 15 == 0? 0 : 1) + 4;
             StringBuilder buf = new StringBuilder(chStr.length() + 1 + eventName.length() + 2 + 10 + 1 + 2 + rows * 80);
```

Comment:
Also adjust `StringBuilder` capacity accordingly? (same for `ByteBufHolder` case below)

and kind of separate to this change but I wonder if it would be better to use `InternalThreadLocalMap#stringBuilder()` in these methods (maybe we don't care much about the allocations though anyhow in cases where this amount of logging is enabled)

Reply:
Will adjust `StringBuilder` capacity, but I'm not sure I fully follow the math. In the sample `shouldLogByteBufDataRead()` unit test, it projects 461 capacity but results in a length of 406. Is it meant to over-estimate or was `appendPrettyHexDump()` updated without updating this logic?